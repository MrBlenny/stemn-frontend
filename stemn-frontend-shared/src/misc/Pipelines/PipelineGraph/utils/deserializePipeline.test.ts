import { deserializePipeline, serializePipeline, createDiagramEngine } from './'
import { pipelineConfigFixture } from '../fixtures'
import { DiagramModel } from 'mrblenny-storm-react-diagrams';

describe('deserializePipeline', () => {
  let diagram: DiagramModel
  it('can deserialize a pipeline', () => {
    const diagramEngine = createDiagramEngine()
    diagram = deserializePipeline(pipelineConfigFixture, diagramEngine)
    expect(diagram)
      .toMatchSnapshot()
  })

  it('can serialize the pipeline', () => {
    expect(serializePipeline(diagram, pipelineConfigFixture.name))
      .toEqual(pipelineConfigFixture)
  })
})
