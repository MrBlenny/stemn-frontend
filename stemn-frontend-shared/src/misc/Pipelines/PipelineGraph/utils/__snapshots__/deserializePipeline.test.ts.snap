// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deserializePipeline can deserialize a pipeline 1`] = `
DiagramModel {
  "gridSize": 0,
  "id": "diagram",
  "links": Object {
    "f9938468-d5fa-4f57-b4d8-4a3a4439800e": DefaultLinkModel {
      "color": undefined,
      "curvyness": undefined,
      "extras": Object {},
      "id": "f9938468-d5fa-4f57-b4d8-4a3a4439800e",
      "labels": Array [],
      "listeners": Object {
        "77539a7d-9879-4182-94c1-96e8ec2c26f3": Object {
          "entityRemoved": [Function],
        },
      },
      "locked": false,
      "parent": [Circular],
      "points": Array [
        PointModel {
          "id": "2ceb02e2-0951-44d9-a463-4405e66f930e",
          "listeners": Object {},
          "locked": false,
          "parent": [Circular],
          "selected": false,
          "type": "default",
          "x": 0,
          "y": 0,
        },
        PointModel {
          "id": "03addf19-6ff9-4d4d-8445-bd7f3e953dda",
          "listeners": Object {},
          "locked": false,
          "parent": [Circular],
          "selected": false,
          "type": "default",
          "x": 0,
          "y": 0,
        },
      ],
      "selected": false,
      "sourcePort": PipelineGraphPortModel {
        "id": "5a27e94e-d61a-4496-8019-9349b319ed58",
        "links": Object {
          "f9938468-d5fa-4f57-b4d8-4a3a4439800e": [Circular],
        },
        "listeners": Object {},
        "locked": false,
        "maximumLinks": 1,
        "name": "5a27e94e-d61a-4496-8019-9349b319ed58",
        "parent": PipelineGraphStepModel {
          "extras": Object {},
          "id": "some_step",
          "listeners": Object {
            "2fff6db7-3fe2-486b-b8e6-4326de33d4ee": Object {
              "entityRemoved": [Function],
            },
          },
          "locked": false,
          "parent": [Circular],
          "ports": Object {
            "5a27e94e-d61a-4496-8019-9349b319ed58": [Circular],
            "dd3bc6f2-058d-4167-bc9f-97152a5b8d63": PipelineGraphPortModel {
              "id": "dd3bc6f2-058d-4167-bc9f-97152a5b8d63",
              "links": Object {},
              "listeners": Object {},
              "locked": false,
              "maximumLinks": 1,
              "name": "dd3bc6f2-058d-4167-bc9f-97152a5b8d63",
              "parent": [Circular],
              "position": "input",
              "selected": false,
              "type": "input",
            },
          },
          "selected": false,
          "type": "some_type",
          "x": 50,
          "y": 50,
        },
        "position": "output",
        "selected": false,
        "type": "output",
      },
      "targetPort": PipelineGraphPortModel {
        "id": "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe",
        "links": Object {
          "f9938468-d5fa-4f57-b4d8-4a3a4439800e": [Circular],
        },
        "listeners": Object {},
        "locked": false,
        "maximumLinks": 1,
        "name": "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe",
        "parent": PipelineGraphStepModel {
          "extras": Object {},
          "id": "some_other_step",
          "listeners": Object {
            "042b22da-610a-479a-95ed-c8c772836e4f": Object {
              "entityRemoved": [Function],
            },
          },
          "locked": false,
          "parent": [Circular],
          "ports": Object {
            "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe": [Circular],
            "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb": PipelineGraphPortModel {
              "id": "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb",
              "links": Object {},
              "listeners": Object {},
              "locked": false,
              "maximumLinks": 1,
              "name": "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb",
              "parent": [Circular],
              "position": "output",
              "selected": false,
              "type": "output",
            },
          },
          "selected": false,
          "type": "some_other_type",
          "x": 200,
          "y": 200,
        },
        "position": "input",
        "selected": false,
        "type": "input",
      },
      "type": "default",
      "width": 3,
    },
  },
  "listeners": Object {},
  "locked": false,
  "nodes": Object {
    "some_other_step": PipelineGraphStepModel {
      "extras": Object {},
      "id": "some_other_step",
      "listeners": Object {
        "042b22da-610a-479a-95ed-c8c772836e4f": Object {
          "entityRemoved": [Function],
        },
      },
      "locked": false,
      "parent": [Circular],
      "ports": Object {
        "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe": PipelineGraphPortModel {
          "id": "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe",
          "links": Object {
            "f9938468-d5fa-4f57-b4d8-4a3a4439800e": DefaultLinkModel {
              "color": undefined,
              "curvyness": undefined,
              "extras": Object {},
              "id": "f9938468-d5fa-4f57-b4d8-4a3a4439800e",
              "labels": Array [],
              "listeners": Object {
                "77539a7d-9879-4182-94c1-96e8ec2c26f3": Object {
                  "entityRemoved": [Function],
                },
              },
              "locked": false,
              "parent": [Circular],
              "points": Array [
                PointModel {
                  "id": "2ceb02e2-0951-44d9-a463-4405e66f930e",
                  "listeners": Object {},
                  "locked": false,
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                  "x": 0,
                  "y": 0,
                },
                PointModel {
                  "id": "03addf19-6ff9-4d4d-8445-bd7f3e953dda",
                  "listeners": Object {},
                  "locked": false,
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                  "x": 0,
                  "y": 0,
                },
              ],
              "selected": false,
              "sourcePort": PipelineGraphPortModel {
                "id": "5a27e94e-d61a-4496-8019-9349b319ed58",
                "links": Object {
                  "f9938468-d5fa-4f57-b4d8-4a3a4439800e": [Circular],
                },
                "listeners": Object {},
                "locked": false,
                "maximumLinks": 1,
                "name": "5a27e94e-d61a-4496-8019-9349b319ed58",
                "parent": PipelineGraphStepModel {
                  "extras": Object {},
                  "id": "some_step",
                  "listeners": Object {
                    "2fff6db7-3fe2-486b-b8e6-4326de33d4ee": Object {
                      "entityRemoved": [Function],
                    },
                  },
                  "locked": false,
                  "parent": [Circular],
                  "ports": Object {
                    "5a27e94e-d61a-4496-8019-9349b319ed58": [Circular],
                    "dd3bc6f2-058d-4167-bc9f-97152a5b8d63": PipelineGraphPortModel {
                      "id": "dd3bc6f2-058d-4167-bc9f-97152a5b8d63",
                      "links": Object {},
                      "listeners": Object {},
                      "locked": false,
                      "maximumLinks": 1,
                      "name": "dd3bc6f2-058d-4167-bc9f-97152a5b8d63",
                      "parent": [Circular],
                      "position": "input",
                      "selected": false,
                      "type": "input",
                    },
                  },
                  "selected": false,
                  "type": "some_type",
                  "x": 50,
                  "y": 50,
                },
                "position": "output",
                "selected": false,
                "type": "output",
              },
              "targetPort": [Circular],
              "type": "default",
              "width": 3,
            },
          },
          "listeners": Object {},
          "locked": false,
          "maximumLinks": 1,
          "name": "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe",
          "parent": [Circular],
          "position": "input",
          "selected": false,
          "type": "input",
        },
        "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb": PipelineGraphPortModel {
          "id": "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb",
          "links": Object {},
          "listeners": Object {},
          "locked": false,
          "maximumLinks": 1,
          "name": "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb",
          "parent": [Circular],
          "position": "output",
          "selected": false,
          "type": "output",
        },
      },
      "selected": false,
      "type": "some_other_type",
      "x": 200,
      "y": 200,
    },
    "some_step": PipelineGraphStepModel {
      "extras": Object {},
      "id": "some_step",
      "listeners": Object {
        "2fff6db7-3fe2-486b-b8e6-4326de33d4ee": Object {
          "entityRemoved": [Function],
        },
      },
      "locked": false,
      "parent": [Circular],
      "ports": Object {
        "5a27e94e-d61a-4496-8019-9349b319ed58": PipelineGraphPortModel {
          "id": "5a27e94e-d61a-4496-8019-9349b319ed58",
          "links": Object {
            "f9938468-d5fa-4f57-b4d8-4a3a4439800e": DefaultLinkModel {
              "color": undefined,
              "curvyness": undefined,
              "extras": Object {},
              "id": "f9938468-d5fa-4f57-b4d8-4a3a4439800e",
              "labels": Array [],
              "listeners": Object {
                "77539a7d-9879-4182-94c1-96e8ec2c26f3": Object {
                  "entityRemoved": [Function],
                },
              },
              "locked": false,
              "parent": [Circular],
              "points": Array [
                PointModel {
                  "id": "2ceb02e2-0951-44d9-a463-4405e66f930e",
                  "listeners": Object {},
                  "locked": false,
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                  "x": 0,
                  "y": 0,
                },
                PointModel {
                  "id": "03addf19-6ff9-4d4d-8445-bd7f3e953dda",
                  "listeners": Object {},
                  "locked": false,
                  "parent": [Circular],
                  "selected": false,
                  "type": "default",
                  "x": 0,
                  "y": 0,
                },
              ],
              "selected": false,
              "sourcePort": [Circular],
              "targetPort": PipelineGraphPortModel {
                "id": "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe",
                "links": Object {
                  "f9938468-d5fa-4f57-b4d8-4a3a4439800e": [Circular],
                },
                "listeners": Object {},
                "locked": false,
                "maximumLinks": 1,
                "name": "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe",
                "parent": PipelineGraphStepModel {
                  "extras": Object {},
                  "id": "some_other_step",
                  "listeners": Object {
                    "042b22da-610a-479a-95ed-c8c772836e4f": Object {
                      "entityRemoved": [Function],
                    },
                  },
                  "locked": false,
                  "parent": [Circular],
                  "ports": Object {
                    "2a94d89f-2a9b-4f5f-85e2-e5cbdc83b5fe": [Circular],
                    "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb": PipelineGraphPortModel {
                      "id": "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb",
                      "links": Object {},
                      "listeners": Object {},
                      "locked": false,
                      "maximumLinks": 1,
                      "name": "d6e930cd-c1bf-4b5a-8cc6-96a854a12bbb",
                      "parent": [Circular],
                      "position": "output",
                      "selected": false,
                      "type": "output",
                    },
                  },
                  "selected": false,
                  "type": "some_other_type",
                  "x": 200,
                  "y": 200,
                },
                "position": "input",
                "selected": false,
                "type": "input",
              },
              "type": "default",
              "width": 3,
            },
          },
          "listeners": Object {},
          "locked": false,
          "maximumLinks": 1,
          "name": "5a27e94e-d61a-4496-8019-9349b319ed58",
          "parent": [Circular],
          "position": "output",
          "selected": false,
          "type": "output",
        },
        "dd3bc6f2-058d-4167-bc9f-97152a5b8d63": PipelineGraphPortModel {
          "id": "dd3bc6f2-058d-4167-bc9f-97152a5b8d63",
          "links": Object {},
          "listeners": Object {},
          "locked": false,
          "maximumLinks": 1,
          "name": "dd3bc6f2-058d-4167-bc9f-97152a5b8d63",
          "parent": [Circular],
          "position": "input",
          "selected": false,
          "type": "input",
        },
      },
      "selected": false,
      "type": "some_type",
      "x": 50,
      "y": 50,
    },
  },
  "offsetX": 0,
  "offsetY": 0,
  "rendered": false,
  "zoom": 100,
}
`;
